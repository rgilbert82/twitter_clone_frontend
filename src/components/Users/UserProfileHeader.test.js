import React from 'react';
import { UserProfileHeader } from '.';
import { shallow } from 'enzyme';

describe(UserProfileHeader, () => {
  let component;

  beforeEach(() => {
    component = shallow(<UserProfileHeader user={{name: 'Ryan', username: 'RG', followers: 2, following: 1}} />);
  });

  it("renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
