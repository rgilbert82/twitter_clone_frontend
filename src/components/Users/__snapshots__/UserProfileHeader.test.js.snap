// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class UserProfileHeader extends _react2.default.Component {
  constructor(props) {
    super(props);
    this.state = {
      isFollowing: false };


    this.createFollow = this.createFollow.bind(this);
    this.deleteFollow = this.deleteFollow.bind(this);
  }

  createFollow() {
    this.props.createFollow(this.props.user.id);
  }

  deleteFollow() {
    this.props.deleteFollow(this.props.user.id);
  }

  render() {
    const tweetPath = \`/users/\${this.props.user.slug}/\`;
    const followingPath = \`/users/\${this.props.user.slug}/following\`;
    const followersPath = \`/users/\${this.props.user.slug}/followers\`;
    let followButton;

    if (!!this.props.isCurrentUserPage && !this.props.isCurrentUserPage()) {
      if (this.props.loggedIn && this.props.isFollowing) {
        followButton = _react2.default.createElement(_reactBootstrap.Button, { onClick: this.deleteFollow, __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this }, 'Unfollow');
      } else if (this.props.loggedIn) {
        followButton = _react2.default.createElement(_reactBootstrap.Button, { onClick: this.createFollow, __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this }, 'Follow');
      }
    }

    return (
      _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 39 }, __self: this },
        _react2.default.createElement('h1', { className: 'section_header', __source: { fileName: _jsxFileName, lineNumber: 40 }, __self: this }, this.props.user.name, _react2.default.createElement('span', { __source: { fileName: _jsxFileName, lineNumber: 40 }, __self: this }, '@', this.props.user.username)),
        _react2.default.createElement('div', { __source: { fileName: _jsxFileName, lineNumber: 41 }, __self: this },
          _react2.default.createElement('div', { className: 'user_stats user_profile_info', __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this },
            _react2.default.createElement('dl', { __source: { fileName: _jsxFileName, lineNumber: 43 }, __self: this },
              _react2.default.createElement('dt', { __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this }, _react2.default.createElement(_reactRouterDom.Link, { to: tweetPath, __source: { fileName: _jsxFileName, lineNumber: 44 }, __self: this }, 'Tweets')),
              _react2.default.createElement('dd', { __source: { fileName: _jsxFileName, lineNumber: 45 }, __self: this }, this.props.user.all_tweet_count)),

            _react2.default.createElement('dl', { __source: { fileName: _jsxFileName, lineNumber: 47 }, __self: this },
              _react2.default.createElement('dt', { __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this }, _react2.default.createElement(_reactRouterDom.Link, { to: followingPath, __source: { fileName: _jsxFileName, lineNumber: 48 }, __self: this }, 'Following')),
              _react2.default.createElement('dd', { __source: { fileName: _jsxFileName, lineNumber: 49 }, __self: this }, this.props.user.following)),

            _react2.default.createElement('dl', { __source: { fileName: _jsxFileName, lineNumber: 51 }, __self: this },
              _react2.default.createElement('dt', { __source: { fileName: _jsxFileName, lineNumber: 52 }, __self: this }, _react2.default.createElement(_reactRouterDom.Link, { to: followersPath, __source: { fileName: _jsxFileName, lineNumber: 52 }, __self: this }, 'Followers')),
              _react2.default.createElement('dd', { __source: { fileName: _jsxFileName, lineNumber: 53 }, __self: this }, this.props.user.followers))),


          _react2.default.createElement('div', { className: 'follow_button', __source: { fileName: _jsxFileName, lineNumber: 56 }, __self: this },
            followButton))));




  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserProfileHeader
    user={
        Object {
            "followers": 2,
            "following": 1,
            "name": "Ryan",
            "username": "RG",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="section_header"
>
          Ryan
          <span>
                    @
                    RG
          </span>
</h1>,
        <div>
          <div
                    className="user_stats user_profile_info"
          >
                    <dl>
                              <dt>
                                        <Link
                                                  replace={false}
                                                  to="/users/undefined/"
                                        >
                                                  Tweets
                                        </Link>
                              </dt>
                              <dd />
                    </dl>
                    <dl>
                              <dt>
                                        <Link
                                                  replace={false}
                                                  to="/users/undefined/following"
                                        >
                                                  Following
                                        </Link>
                              </dt>
                              <dd>
                                        1
                              </dd>
                    </dl>
                    <dl>
                              <dt>
                                        <Link
                                                  replace={false}
                                                  to="/users/undefined/followers"
                                        >
                                                  Followers
                                        </Link>
                              </dt>
                              <dd>
                                        2
                              </dd>
                    </dl>
          </div>
          <div
                    className="follow_button"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Ryan",
            <span>
              @
              RG
</span>,
          ],
          "className": "section_header",
        },
        "ref": null,
        "rendered": Array [
          "Ryan",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "@",
                "RG",
              ],
            },
            "ref": null,
            "rendered": Array [
              "@",
              "RG",
            ],
            "type": "span",
          },
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="user_stats user_profile_info"
>
              <dl>
                            <dt>
                                          <Link
                                                        replace={false}
                                                        to="/users/undefined/"
                                          >
                                                        Tweets
                                          </Link>
                            </dt>
                            <dd />
              </dl>
              <dl>
                            <dt>
                                          <Link
                                                        replace={false}
                                                        to="/users/undefined/following"
                                          >
                                                        Following
                                          </Link>
                            </dt>
                            <dd>
                                          1
                            </dd>
              </dl>
              <dl>
                            <dt>
                                          <Link
                                                        replace={false}
                                                        to="/users/undefined/followers"
                                          >
                                                        Followers
                                          </Link>
                            </dt>
                            <dd>
                                          2
                            </dd>
              </dl>
</div>,
            <div
              className="follow_button"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <dl>
                  <dt>
                                    <Link
                                                      replace={false}
                                                      to="/users/undefined/"
                                    >
                                                      Tweets
                                    </Link>
                  </dt>
                  <dd />
</dl>,
                <dl>
                  <dt>
                                    <Link
                                                      replace={false}
                                                      to="/users/undefined/following"
                                    >
                                                      Following
                                    </Link>
                  </dt>
                  <dd>
                                    1
                  </dd>
</dl>,
                <dl>
                  <dt>
                                    <Link
                                                      replace={false}
                                                      to="/users/undefined/followers"
                                    >
                                                      Followers
                                    </Link>
                  </dt>
                  <dd>
                                    2
                  </dd>
</dl>,
              ],
              "className": "user_stats user_profile_info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <dt>
                      <Link
                                            replace={false}
                                            to="/users/undefined/"
                      >
                                            Tweets
                      </Link>
</dt>,
                    <dd />,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/users/undefined/"
>
                        Tweets
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Tweets",
                        "replace": false,
                        "to": "/users/undefined/",
                      },
                      "ref": null,
                      "rendered": "Tweets",
                      "type": [Function],
                    },
                    "type": "dt",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "dd",
                  },
                ],
                "type": "dl",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <dt>
                      <Link
                                            replace={false}
                                            to="/users/undefined/following"
                      >
                                            Following
                      </Link>
</dt>,
                    <dd>
                      1
</dd>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/users/undefined/following"
>
                        Following
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Following",
                        "replace": false,
                        "to": "/users/undefined/following",
                      },
                      "ref": null,
                      "rendered": "Following",
                      "type": [Function],
                    },
                    "type": "dt",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 1,
                    },
                    "ref": null,
                    "rendered": 1,
                    "type": "dd",
                  },
                ],
                "type": "dl",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <dt>
                      <Link
                                            replace={false}
                                            to="/users/undefined/followers"
                      >
                                            Followers
                      </Link>
</dt>,
                    <dd>
                      2
</dd>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Link
                        replace={false}
                        to="/users/undefined/followers"
>
                        Followers
</Link>,
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Followers",
                        "replace": false,
                        "to": "/users/undefined/followers",
                      },
                      "ref": null,
                      "rendered": "Followers",
                      "type": [Function],
                    },
                    "type": "dt",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": 2,
                    },
                    "ref": null,
                    "rendered": 2,
                    "type": "dd",
                  },
                ],
                "type": "dl",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "follow_button",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="section_header"
>
            Ryan
            <span>
                        @
                        RG
            </span>
</h1>,
          <div>
            <div
                        className="user_stats user_profile_info"
            >
                        <dl>
                                    <dt>
                                                <Link
                                                            replace={false}
                                                            to="/users/undefined/"
                                                >
                                                            Tweets
                                                </Link>
                                    </dt>
                                    <dd />
                        </dl>
                        <dl>
                                    <dt>
                                                <Link
                                                            replace={false}
                                                            to="/users/undefined/following"
                                                >
                                                            Following
                                                </Link>
                                    </dt>
                                    <dd>
                                                1
                                    </dd>
                        </dl>
                        <dl>
                                    <dt>
                                                <Link
                                                            replace={false}
                                                            to="/users/undefined/followers"
                                                >
                                                            Followers
                                                </Link>
                                    </dt>
                                    <dd>
                                                2
                                    </dd>
                        </dl>
            </div>
            <div
                        className="follow_button"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Ryan",
              <span>
                @
                RG
</span>,
            ],
            "className": "section_header",
          },
          "ref": null,
          "rendered": Array [
            "Ryan",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "@",
                  "RG",
                ],
              },
              "ref": null,
              "rendered": Array [
                "@",
                "RG",
              ],
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="user_stats user_profile_info"
>
                <dl>
                                <dt>
                                                <Link
                                                                replace={false}
                                                                to="/users/undefined/"
                                                >
                                                                Tweets
                                                </Link>
                                </dt>
                                <dd />
                </dl>
                <dl>
                                <dt>
                                                <Link
                                                                replace={false}
                                                                to="/users/undefined/following"
                                                >
                                                                Following
                                                </Link>
                                </dt>
                                <dd>
                                                1
                                </dd>
                </dl>
                <dl>
                                <dt>
                                                <Link
                                                                replace={false}
                                                                to="/users/undefined/followers"
                                                >
                                                                Followers
                                                </Link>
                                </dt>
                                <dd>
                                                2
                                </dd>
                </dl>
</div>,
              <div
                className="follow_button"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <dl>
                    <dt>
                                        <Link
                                                            replace={false}
                                                            to="/users/undefined/"
                                        >
                                                            Tweets
                                        </Link>
                    </dt>
                    <dd />
</dl>,
                  <dl>
                    <dt>
                                        <Link
                                                            replace={false}
                                                            to="/users/undefined/following"
                                        >
                                                            Following
                                        </Link>
                    </dt>
                    <dd>
                                        1
                    </dd>
</dl>,
                  <dl>
                    <dt>
                                        <Link
                                                            replace={false}
                                                            to="/users/undefined/followers"
                                        >
                                                            Followers
                                        </Link>
                    </dt>
                    <dd>
                                        2
                    </dd>
</dl>,
                ],
                "className": "user_stats user_profile_info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <dt>
                        <Link
                                                replace={false}
                                                to="/users/undefined/"
                        >
                                                Tweets
                        </Link>
</dt>,
                      <dd />,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/users/undefined/"
>
                          Tweets
</Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Tweets",
                          "replace": false,
                          "to": "/users/undefined/",
                        },
                        "ref": null,
                        "rendered": "Tweets",
                        "type": [Function],
                      },
                      "type": "dt",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "dd",
                    },
                  ],
                  "type": "dl",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <dt>
                        <Link
                                                replace={false}
                                                to="/users/undefined/following"
                        >
                                                Following
                        </Link>
</dt>,
                      <dd>
                        1
</dd>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/users/undefined/following"
>
                          Following
</Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Following",
                          "replace": false,
                          "to": "/users/undefined/following",
                        },
                        "ref": null,
                        "rendered": "Following",
                        "type": [Function],
                      },
                      "type": "dt",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 1,
                      },
                      "ref": null,
                      "rendered": 1,
                      "type": "dd",
                    },
                  ],
                  "type": "dl",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <dt>
                        <Link
                                                replace={false}
                                                to="/users/undefined/followers"
                        >
                                                Followers
                        </Link>
</dt>,
                      <dd>
                        2
</dd>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Link
                          replace={false}
                          to="/users/undefined/followers"
>
                          Followers
</Link>,
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "children": "Followers",
                          "replace": false,
                          "to": "/users/undefined/followers",
                        },
                        "ref": null,
                        "rendered": "Followers",
                        "type": [Function],
                      },
                      "type": "dt",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": 2,
                      },
                      "ref": null,
                      "rendered": 2,
                      "type": "dd",
                    },
                  ],
                  "type": "dl",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "follow_button",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
